### DISCUSSION TITLE
Meeting Notes: 2024-07-29

### DISCUSSION THREAD

**Discussion by JonathanStarup** [flix/flix#8486] [General]
* Refactor away imports
    * object calls, constructors, and static calls with the IO effect or pure (use `unsafe X`)
    * Matt makes a list of files to do 
* Null and arrays should be supported during this week
* Structs
    * should it have a scheme? yes, but not a function like `Enum.tag` has
        * Scheme.instantiate should returns its free var substitution, the struct typing can then apply that substitution to the field type
* merge FastSetUnifcation
* Java classes as effects? good idea - roundable to IO

**Comment by chanattan** [comment]
>     * Null and arrays should be supported during this week
- Null values should already be supported, some examples are provided in `Test.Exp.Jvm.InvokeMethod2.Overload.Null` for information.
- Arrays are almost supported, it lacks full support with regions and there remains one error that I'm figuring out with Matt. Tests are not implemented yet.

**Comment by JonathanStarup** [comment]
moved to https://github.com/mlutze/flix-meetings
